raiden-cli/src/routes/channels.ts

```diff
@@ -1,60 +1,97 @@
-import { Router, Request, Response } from 'express';
+import { Router, Request, Response, NextFunction } from 'express';
 import { first } from 'rxjs/operators';
+import { ErrorCodes, RaidenError } from 'raiden-ts';
 import { Cli } from '../types';
-import { validateAddressParameter } from '../utils/validation';
+import {
+  validateAddressParameter,
+  isInvalidParameterError,
+  isTransactionWouldFailError,
+} from '../utils/validation';
 import {
   flattenChannelDictionary,
   transformSdkChannelFormatToApi,
   filterChannels,
 } from '../utils/channels';
 
+function isConflictError(error: RaidenError): boolean {
+  return (
+    [ErrorCodes.RDN_UNKNOWN_TOKEN_NETWORK, ErrorCodes.CNL_INVALID_STATE].includes(error.message) ||
+    isTransactionWouldFailError(error)
+  );
+}
+
+async function getAllChannels(this: Cli, _request: Request, response: Response) {
+  const channelDictionary = await this.raiden.channels$.pipe(first()).toPromise();
+  const channelList = flattenChannelDictionary(channelDictionary);
+  const formattedChannels = channelList.map((channel) => transformSdkChannelFormatToApi(channel));
+  response.json(formattedChannels);
+}
+
+async function getChannelsForToken(this: Cli, request: Request, response: Response) {
+  const channelDictionary = await this.raiden.channels$.pipe(first()).toPromise();
+  const channelList = flattenChannelDictionary(channelDictionary);
+  const filteredChannels = filterChannels(channelList, request.params.tokenAddress);
+  const formattedChannels = filteredChannels.map((channel) =>
+    transformSdkChannelFormatToApi(channel),
+  );
+  response.json(formattedChannels);
+}
+
+async function getChannelsForTokenAndPartner(this: Cli, request: Request, response: Response) {
+  const channelDictionary = await this.raiden.channels$.pipe(first()).toPromise();
+  const channel = channelDictionary[request.params.tokenAddress]?.[request.params.partnerAddress];
+
+  if (channel === undefined) {
```

Comment:
```suggestion
  if (channel) {
```
`channel === undefined` would be the `else` condition. Also, for objects, no need to be explicit about `undefined`, as simple boolean conditional check is enough unless you'd have other falsy values in your union (which isn't the case here)

Reply:
Oh damn. See? We need some kind of tests I can run quickly if I do such code rework. I can't do this manually all the time. :expressionless:

Thanks that you caught that!

Reply:
:heavy_check_mark: